From 4eaf85ad2669dd8b527c36de3eb3c8d2b85d3544 Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Thu, 10 Mar 2022 14:11:13 +0200
Subject: [PATCH] [SW-76562] gaudi2: change axi drain timeout

PSOC AXI drain timeout increased to meet profiler team requirements

As per VLSI team drain time out of 0x2000 corresponds to 16 microsec
and this value should be enough for PSOC AXI drain time out
But profiler team is seeing device hangs when there is burst of trace
on HBM. If the timeout value is changed to 0xF0000 no hangs are seen
So changing the timeout value to 0xF0000 till actual hardware issue
is found

Signed-off-by: Sailaja <sbandarupalli@habana.ai>
Change-Id: I17776086f02dc968b817f25f5d708c2837068ef9
(cherry picked from commit f0ddb08f92fb76b4122c63207c9cd6e5b8a4ec5b)
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 52747a6..cd84b6d 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -1079,7 +1079,15 @@
 	clr_flr_reset();
 }
 
-#define AXI_DRAIN_TIMEOUT	0x2000
+/*
+ * As per VLSI team drain time out of 0x2000 corresponds to 16 micro seconds
+ * and this value should be enough for AXI drain time out
+ * Profiler team is seeing device hangs when there is burst of trace on HBM
+ * If the timeout value is changed to 0xF0000 no hangs are seen
+ * So changing the timeout value to 0xF0000 till actual hardware issue found
+ * TODO : Set correct AXI_DRAIN_TIMEOUT value
+ */
+#define AXI_DRAIN_TIMEOUT	0xF0000
 
 void gaudi2_axi_drain_ena(void)
 {
